sdk/eventhub/azure-eventhubs/azure/eventhub/_consumer_producer_mixin.py

```diff
@@ -7,12 +7,31 @@
 import logging
 import time
 
-from uamqp import errors
+from uamqp import errors, constants, compat
 from azure.eventhub.error import EventHubError, _handle_exception
 
 log = logging.getLogger(__name__)
 
 
+def _retry_decorator(to_be_wrapped_func):
+    def wrapped_func(self, *args, **kwargs):
+        timeout = kwargs.get("timeout", None)
```

Comment:
`timeout = kwargs.pop("timeout", None)` (and remove the pop below)?

Reply:
good idea, I'll fix this.